api/src/main/java/org/apache/iceberg/util/BinaryUtil.java

```diff
@@ -50,6 +50,22 @@ public static ByteBuffer truncateBinary(ByteBuffer input, int length) {
     return ByteBuffer.wrap(array);
   }
 
+  /**
+   * Truncates the input byte buffer to the given length.
+   *
+   * <p>This skips input validation, and avoids copying the backing bytes if possible for efficiency
```

Comment:
Why does this say "if possible"? This always returns a duplicate of the original buffer.

The distinction between this and `truncateBinary` is that this doesn't copy the data. You can remove any wording about efficiency or input validation. The buffer handles validation and will throw IllegalArgumentException if needed.

Reply:
The main part remaining about skipping input validation is that `width` isn't validated to be non-negative. I can leave that out though if desired.

Reply:
Yep. The buffer handles validation.